compiler/qsc_qasm3/src/parser/stmt.rs

```diff
@@ -57,7 +58,35 @@ pub(super) fn parse(s: &mut ParserContext) -> Result<Box<Stmt>> {
         Box::new(decl)
     } else if let Some(include) = opt(s, parse_include)? {
         Box::new(include)
-    } else if let Some(decl) = opt(s, parse_local)? {
+    } else if let Some(ty) = opt(s, scalar_or_array_type)? {
+        if matches!(s.peek().kind, TokenKind::Identifier) {
+            Box::new(parse_non_constant_classical_decl(s, ty, lo)?)
+        } else {
+            token(s, TokenKind::Open(Delim::Paren))?;
+            let arg = expr::expr(s)?;
+            recovering_token(s, TokenKind::Close(Delim::Paren));
+            let cast_expr = Expr {
+                span: s.span(lo),
+                kind: Box::new(ExprKind::Cast(Cast {
+                    span: s.span(lo),
+                    r#type: ty,
+                    arg,
+                })),
+            };
+            Box::new(StmtKind::ExprStmt(ExprStmt {
+                span: s.span(lo),
+                expr: cast_expr,
+            }))
+        }
```

Comment:
This part is missing the call to `parse_expression_stmt` with the generated cast, and the spans should come from the already parsed type:

```suggestion
            let cast_expr = Expr {
                span: s.span(ty.span().lo),
                kind: Box::new(ExprKind::Cast(Cast {
                    span: s.span(ty.span().lo),
                    r#type: ty,
                    arg,
                })),
            };
            Box::new(StmtKind::ExprStmt(parse_expression_stmt(
                s,
                Some(cast_expr),
            )?))
        }
```